errdefs/http_helpers.go

```diff
@@ -170,3 +175,24 @@ func statusCodeFromDistributionError(err error) int {
 	}
 	return http.StatusInternalServerError
 }
+
+// statusCodeFromContainerdError returns status code for containerd errors when
+// consumed directly (not through gRPC)
+func statusCodeFromContainerdError(err error) int {
```

Comment:
With containerd errors they only need to be unwrapped once with `errors.Cause` then a normal switch could be used. This is fine though for now, hopefully we can make the techniques more consistent so they can all be checked together in the future (for example maybe docker's version of the `Is*` functions can just also check containerd errors there too).

Reply:
Yes, we should look at integrating them (if they all can be mapped). As to unwrapping; it's possible (although not the case currently) that errors are wrapped multiple times, so I guess recursively unwrapping is still ok to do